### PR TITLE
[eventd] Fix eventd UT flakiness

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

Fix flakiness of eventd UT - run sub after capture service starts

##### Work item tracking
- Microsoft ADO **(number only)**:25650744

#### How I did it

Run sub socket after capture socket is initialized

#### How to verify it

Pipeline

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-eventd/tests/eventd_ut.cpp [hunk 1]
```diff
@@ -595,6 +601,7 @@ TEST(eventd, service)
     }
 
     thread thread_service(&run_eventd_service);
+    this_thread::sleep_for(chrono::milliseconds(1000));
```
Comment (ganglyu) [inline src/sonic-eventd/tests/eventd_ut.cpp:604]: why do you add sleep here?

Comment (zbud-msft) [reply src/sonic-eventd/tests/eventd_ut.cpp:604]: This sleep is to give time for the run_eventd_service thread to establish zmq proxy and to start the capture service.

**Review by ganglyu — APPROVED** [review]

**Comment by zbud-msft** [comment]
@StormLiangMS Can you please review? I have tested this issue on official build pipeline and I am no longer seeing any flakiness in 12+ consecutive runs.

File: src/sonic-eventd/tests/eventd_ut.cpp [hunk 2]
```diff
@@ -625,7 +632,7 @@ TEST(eventd, service)
         run_pub(mock_pub, wr_source, wr_evts);
 
         /* Published events must have been captured. Give a pause, to ensure sent. */
-        this_thread::sleep_for(chrono::milliseconds(200));
+        this_thread::sleep_for(chrono::milliseconds(1000));
```
Comment (qiluo-msft) [inline src/sonic-eventd/tests/eventd_ut.cpp:635]: >1000 [](http://example.com/codeflow?start=52&length=4)

Is this a critical fix? How do you pick the magic number?

File: src/sonic-eventd/tests/eventd_ut.cpp (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-eventd/tests/eventd_ut.cpp:604]: Is it possible to wait a single, do not use magic number.

File: src/sonic-eventd/tests/eventd_ut.cpp (hunk 2 above)
Comment (renukamanavalan) [reply src/sonic-eventd/tests/eventd_ut.cpp:635]: ZMQ is async. No signal indication. It silently drops when not good.

File: src/sonic-eventd/tests/eventd_ut.cpp (hunk 1 above)
Comment (zbud-msft) [reply src/sonic-eventd/tests/eventd_ut.cpp:604]: We sleep for 1 second to wait for zmq proxy to be established and capture service to start, which can take up to 1 second: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-eventd/src/eventd.cpp#L549

Comment (qiluo-msft) [reply src/sonic-eventd/tests/eventd_ut.cpp:604]: Thanks for pointing out where 1s comes from! If someone change the eventd.cpp later, how can we make the testcase is also updated?

File: src/sonic-eventd/tests/eventd_ut.cpp [hunk 3]
```diff
@@ -624,15 +631,32 @@ TEST(eventd, service)
...
 
-        /* Published events must have been captured. Give a pause, to ensure sent. */
-        this_thread::sleep_for(chrono::milliseconds(200));
+        int max_polling_duration = 2000;
+        int polling_interval = 100;
+        auto poll_start_ts = chrono::steady_clock::now();
+
+        while(true) {
+            auto current_ts = chrono::steady_clock::now();
+	    if(chrono::duration_cast<chrono::milliseconds>(current_ts - poll_start_ts).count() >= max_polling_duration) {
```
Comment (qiluo-msft) [inline src/sonic-eventd/tests/eventd_ut.cpp:640]: >	     [](http://example.com/codeflow?start=0&length=5)

Mix using tabs (several lines in the new code) and spaces.

Comment (zbud-msft) [reply src/sonic-eventd/tests/eventd_ut.cpp]: Done

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**